cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -216,68 +274,42 @@ func (scaleSet *ScaleSet) updateVMSSCapacity(future *azure.Future) {
 		}
 	}()
 
-	ctx, cancel := getContextWithCancel()
+	ctx, cancel := getContextWithTimeout(asyncContextTimeout)
 	defer cancel()
 
 	klog.V(3).Infof("Calling virtualMachineScaleSetsClient.WaitForCreateOrUpdateResult(%s)", scaleSet.Name)
 	httpResponse, err := scaleSet.manager.azClient.virtualMachineScaleSetsClient.WaitForCreateOrUpdateResult(ctx, future, scaleSet.manager.config.ResourceGroup)
 
 	isSuccess, err := isSuccessHTTPResponse(httpResponse, err)
 	if isSuccess {
-		klog.V(3).Infof("virtualMachineScaleSetsClient.WaitForCreateOrUpdateResult(%s) success", scaleSet.Name)
-		scaleSet.invalidateInstanceCache()
+		klog.V(3).Infof("waitForCreateOrUpdateInstances(%s) success", scaleSet.Name)
 		return
 	}
 
-	klog.Errorf("virtualMachineScaleSetsClient.WaitForCreateOrUpdateResult - updateVMSSCapacity for scale set %q failed: %v", scaleSet.Name, err)
+	klog.Errorf("waitForCreateOrUpdateInstances(%s) failed, err: %v", scaleSet.Name, err)
 }
 
 // SetScaleSetSize sets ScaleSet size.
-func (scaleSet *ScaleSet) SetScaleSetSize(size int64) error {
-	scaleSet.sizeMutex.Lock()
-	defer scaleSet.sizeMutex.Unlock()
-
+func (scaleSet *ScaleSet) SetScaleSetSize(size int64, delta int) error {
 	vmssInfo, err := scaleSet.getVMSSFromCache()
 	if err != nil {
 		klog.Errorf("Failed to get information for VMSS (%q): %v", scaleSet.Name, err)
 		return err
 	}
 
-	// Update the new capacity to cache.
-	vmssSizeMutex.Lock()
-	vmssInfo.Sku.Capacity = &size
-	vmssSizeMutex.Unlock()
-
-	// Compose a new VMSS for updating.
-	op := compute.VirtualMachineScaleSet{
-		Name:     vmssInfo.Name,
-		Sku:      vmssInfo.Sku,
-		Location: vmssInfo.Location,
-	}
+	requiredInstances := delta
 
-	if vmssInfo.ExtendedLocation != nil {
```

Comment:
Not present on older versions. But justified(?) due to fork code differences, see `createOrUpdateInstances()` for after-fork location.